pkg/envoy/sds/response.go

```diff
@@ -12,155 +10,39 @@ import (
-
-	return certs
-}
-
-// getServiceCertSecret creates the struct with certificates for the service, which the
-// connected Envoy proxy belongs to.
-func getServiceCertSecret(cert *certificate.Certificate, name string) (*xds_auth.Secret, error) {
-	secret := &xds_auth.Secret{
-		// The Name field must match the tls_context.common_tls_context.tls_certificate_sds_secret_configs.name in the Envoy yaml config
-		Name: name,
-		Type: &xds_auth.Secret_TlsCertificate{
-			TlsCertificate: &xds_auth.TlsCertificate{
-				CertificateChain: &xds_core.DataSource{
-					Specifier: &xds_core.DataSource_InlineBytes{
-						InlineBytes: cert.GetCertificateChain(),
-					},
-				},
-				PrivateKey: &xds_core.DataSource{
-					Specifier: &xds_core.DataSource_InlineBytes{
-						InlineBytes: cert.GetPrivateKey(),
-					},
-				},
-			},
-		},
-	}
-	return secret, nil
-}
-
-func (s *sdsImpl) getRootCert(cert *certificate.Certificate, sdscert secrets.SDSCert) (*xds_auth.Secret, error) {
-	secret := &xds_auth.Secret{
-		// The Name field must match the tls_context.common_tls_context.tls_certificate_sds_secret_configs.name
-		Name: sdscert.String(),
-		Type: &xds_auth.Secret_ValidationContext{
-			ValidationContext: &xds_auth.CertificateValidationContext{
-				TrustedCa: &xds_core.DataSource{
-					Specifier: &xds_core.DataSource_InlineBytes{
-						InlineBytes: cert.GetTrustedCAs(),
-					},
-				},
-			},
-		},
-	}
-
-	// SAN validation should not be performed by the root validation certificate used by the upstream server
-	// to validate a downstream client. This is because of the following:
-	// 1. SAN validation is already performed by the RBAC filter on the inbound listener's filter chain (using
-	//    network RBAC filter) and each HTTP route in the inbound route ocnfiguration (using HTTP RBAC per route).
-	// 2. The same root validation certificate is used to validate both in-mesh and ingress downstreams.
-	//
-	// For these reasons, we only perform SAN validation of peer certificates on downstream clients (ie. outbound SAN
-	// validation).
-	if sdscert.CertType == secrets.RootCertTypeForMTLSInbound {
-		return secret, nil
-	}
+	builder.SetTrustDomain(certManager.GetTrustDomain())
 
-	// For the outbound certificate validation context, the SANs needs to match the list of service identities
-	// corresponding to the upstream service. This means, if the sdscert.Name points to service 'X',
-	// the SANs for this certificate should correspond to the service identities of 'X'.
-	meshSvc, err := sdscert.GetMeshService()
-	if err != nil {
-		log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrGettingMeshService)).
-			Msgf("Error unmarshalling upstream service for outbound cert %s", sdscert)
-		return nil, err
-	}
-	svcIdentitiesInCertRequest := s.meshCatalog.ListServiceIdentitiesForService(*meshSvc)
+	// Lists service identities for certs in request
+	builder.SetListServiceIdentitiesForService(meshCatalog.ListServiceIdentitiesForService)
```

Comment:
instead of having a setter that takes in a method, we want to remove all API calls from the builder, and only set data.

What we can do instead is maintain a `map[service.MeshService][]identity.ServiceIdentity` on the builder.

And we can set the data from the response.go's `NewResponse` function.